File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/tv/episodegroup/model/TvEpisodeGroup.java
package dev.reuss.tmdb.domain.tv.episodegroup.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Episode group inside a TV episode group.
 *
 * @param id       group id
 * @param name     group name
 * @param order    group order
 * @param episodes grouped episodes
 * @param locked   whether the group is locked
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvEpisodeGroup(
        String id,
        String name,
        int order,
        List<TvEpisodeGroupEpisode> episodes,
        boolean locked
) implements TmdbModel {

    public TvEpisodeGroup {
        episodes = TmdbCollections.list(episodes);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/tv/episodegroup/model/TvEpisodeGroup.kt
package dev.reuss.tmdb.domain.tv.episodegroup.model

import com.fasterxml.jackson.annotation.JsonProperty

import dev.reuss.tmdb.common.TmdbModel

/**
 * Episode group inside a TV episode group.
 *
 * @property id group id
 * @property name group name
 * @property order group order
 * @property episodes grouped episodes
 * @property locked whether the group is locked
 */
@JvmRecord
data class TvEpisodeGroup(
    @all:JsonProperty("id")
    val id: String?,

    @all:JsonProperty("name")
    val name: String?,

    @all:JsonProperty("order")
    val order: Int,

    @all:JsonProperty("episodes")
    val episodes: List<TvEpisodeGroupEpisode> = emptyList(),

    @all:JsonProperty("locked")
    val locked: Boolean
) : TmdbModel
